homeassistant/components/shelly/__init__.py

```diff
@@ -206,6 +210,10 @@ def mac(self):
 
     async def async_setup(self):
         """Set up the wrapper."""
+
+        if self.hass.is_stopping:
```

Comment:
In what cases are we setting up a wrapper during shutdown?

Reply:
I was not able to identify the exact case, but based on trace error this was one of the 2 func creating issues.
After the change I was no more able to recreate.

Reply:
Will skip for now